packages/web/src/components/userprofile/EditUserForm.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import {
+  Button,
+  ButtonGroup,
+  FormControl,
+  FormLabel,
+  FormHelperText,
+  Input,
+  useDisclosure,
+  Alert,
+  AlertIcon,
+  AlertDescription,
+  CloseButton,
+} from '@chakra-ui/react';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+import { User } from '../../pages/user/[uid]';
+
+type EditFormValues = yup.InferType<typeof editFormSchema>;
+const editFormSchema = yup.object({
+  name: yup.string().trim().required('Name is required.'),
+  pronouns: yup.string(),
+  schoolName: yup.string(),
+});
+
+export interface EditUserProps {
+  setEditToggle(editToggle: boolean): void;
+  setUser(user: User): void;
+  user: {
+    id: string;
+    name: string;
+    pronouns?: string;
+    schoolName?: string;
+  };
+}
+
+export const EditUserForm: React.FC<EditUserProps> = (props: EditUserProps) => {
+  const [errorMessage, setErrorMessage] = React.useState(false);
+
+  const formik = useFormik<EditFormValues>({
+    initialValues: {
+      name: props.user.name,
+      pronouns: props.user.pronouns,
+      schoolName: props.user.schoolName,
+    },
+    validationSchema: editFormSchema,
+    onSubmit: async (data) => {
+      try {
+        const res = await fetch(`/api/users/${props.user.id}`, {
+          method: 'PATCH',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(data),
+        });
+        if (res.ok) {
+          props.setUser(await res.json());
+          props.setEditToggle(false);
+        } else {
+          setErrorMessage(true);
+        }
+      } catch {
+        setErrorMessage(true);
+      }
+    },
+  });
+
+  return (
+    <div>
```

Comment:
Please use a Chakra element here instead of a `div`. It's hard to figure out how this will look as just code, so let me know if you need help with styling